crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -889,14 +931,42 @@ where
                             self.blockchain.set_canonical_head(max_header);
                         }
 
-                        // Update the state and hashes of the blockchain tree if possible.
-                        match self.restore_tree_if_possible(current_state) {
-                            Ok(_) => self.sync_state_updater.update_sync_state(SyncState::Idle),
-                            Err(error) => {
-                                error!(target: "consensus::engine", ?error, "Error restoring blockchain tree");
-                                return Some(Err(error.into()))
-                            }
-                        };
+                        // TODO: figure out how to make this less complex:
+                        // restore_tree_if_possible will run the pipeline if the current_state head
```

Comment:
I still feel that `restore` is wrong wording here, I associate this with `repair` which is not the case here.

so this always derails my train of thought, but perhaps it's just me.

Reply:
maybe this is better described as `reset`, since it sets the state of the tree, or runs the pipeline to the given block